component/src/main/java/org/wso2/extension/siddhi/execution/geo/GeoCoordinateStreamFunctionProcessor.java

```diff
@@ -0,0 +1,216 @@
+ */
+@Extension(
+        name = "geocoordinate",
+        namespace = "geo",
+        description = "geocoordinate stream function returns the longitude and latitude" +
+                " values of a location which is related to the given IPV4 or IPV6 address.",
+        parameters = {
+                @Parameter(
+                        name = "ip",
+                        description = "The IP address that the user need " +
+                                "to get the longitude and latitude",
+                        type = {DataType.STRING})
+        },
+        returnAttributes = {
+                @ReturnAttribute(
+                        name = "longitude",
+                        description = "The longitude of the location " +
+                                "which is related to the given IP", type = DataType.DOUBLE
+                ),
+                @ReturnAttribute(
+                        name = "latitude",
+                        description = "The latitude of the location " +
+                                "which is related to the given IP", type = DataType.DOUBLE
+                )
+        },
+        systemParameter = {
+                @SystemParameter(
+                        name = "apiurl",
+                        description = "ipInfoDB(https://www.ipinfodb.com/) provides an API to" +
+                                "get IP information from their IP address geolocation database." +
+                                "This API provides an url to get the information based on IP address",
+                        defaultValue = "N/A",
+                        possibleParameters = "N/A"
+                )
+        },
+        examples = @Example(
+                description = "This will return the longitude and latitude of the given IPV4 or IPV6 address. " +
+                        "So the results for the geocoordinate(95.31.18.119) are 55.7522, 37.6156",
+                syntax = "define stream IpStream(ip string); " +
+                        "from IpStream#geo:geocoordinate(ip) " +
+                        "select latitude, longitude " +
+                        "insert into outputStream;")
+)
+
+public class GeoCoordinateStreamFunctionProcessor extends StreamFunctionProcessor {
+    private static GeoCoordinateResolver geoCoordinateResolverImpl;
+    private static final String DEFAULT_GEOCOORDINATE_RESOLVER_CLASSNAME =
+            "org.wso2.extension.siddhi.execution.geo.internal.impl.APIBasedGeoCoordinateResolver";
+    private static AtomicBoolean isExtensionConfigInitialized = new AtomicBoolean(false);
+
+    /**
+     * The process method used when more than one function parameters are provided
+     *
+     * @param data the data values for the function parameters
+     * @return the data for additional output attributes introduced by the function
+     */
+    @Override
+    protected Object[] process(Object[] data) {
+        return new Object[0];
```

Comment:
Why are we returning `new Object[0]` here?

Reply:
Fixed with commit : 31d680148c8227446cf78aea89c5867e60a10efd

Reply:
I think we do not need to implement this method. So, IMO we can return `null` here.

Reply:
Fixed with commit : 1b04911ccdaea7c209d8e852bb3d3d29b10e0570